v-next/core/src/config.ts

```diff
@@ -57,11 +58,11 @@ export function overrideTask(
 /**
  * Defines a global parameter.
  */
-export function globalParameter(options: {
+export function globalParameter<T extends ParameterType>(options: {
   name: string;
   description: string;
-  parameterType: ParameterType;
-  defaultValue: any;
+  parameterType?: T;
+  defaultValue: ParameterTypeToValueType<T>;
```

Comment:
@alcuadrado I've added the type validation we use in the task parameters, I think it makes sense to have it here as well.